docs/dev/benchmark.rst

```diff
@@ -0,0 +1,123 @@
+| hardware\_config      | "mbp13inch2019"                                                                                                                                                              | hardware     | descriptor of target hardware environment, optionally json                   |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+------------------------------------------------------------------------------+
+| execution\_config     | {“number”: 1, “repeat”: 10, “min\_repeat\_ms”, 0}                                                                                                                            | statistics   | workload execution parameters, optional                                      |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+------------------------------------------------------------------------------+
+| statistics            | {“runtime\_ms\_mean”: 12,“runtime\_ms\_std”: 12}                                                                                                                             | statistics   | all metrics, other optional fields ``binary_size``, ``compile_time``, etc.   |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+------------------------------------------------------------------------------+
+
+
+
+Storage format
+~~~~~~~~~~~~~~
+Currently we're prototyping benchmark data as JSON objects for extensibility and convenience, especially in early versions of the schema. However, as we scale up benchmark aggregation and stabilize parameters, we anticipate switching to a columnar format, such as Arrow or Parquet.
+
+Here is sample data encoded as JSON:
+
+::
+
+  {
+    "compilation_config":{
+      "relay_opt_lvl":3,
+      "tvm_target":"llvm -mcpu=cascadelake"
+    },
+    "engine":"tvm",
+    "engine_config":{
+      "llvm":"llvm-8"
+    },
+    "engine_version":"3486e2c2cf9d0c7c20853f3503528444ce824c1b",
+    "hardware_config":"c2-standard-16",
+    "metadata":{
+      "docker_tag":"tvmai/ci-gpu:v0.53"
+    },
+    "schema_version":0.1,
+    "software_config":{
+      "os":"ubuntu:18.04",
+      "pip":{
+        "docker":"4.1.0",
+        "gitpython":"3.0.4",
+        "numpy":"1.17.4",
+        "onnx":"1.6.0"
+      }
+    },
+    "statistics":{
+      "compile_ms":3030.268669128418,
+      "runtime_ms_mean":42.473589888888895,
+      "runtime_ms_std":2.184165809925635
```

Comment:
Was there any discussion around storing raw performance data vs the simplified form here ? 

 I've found that keeping the raw data allows one to process it in other ways in terms of { iteration: 1 runtime: <seconds> .... } ? The statistics could well be calculated by the visualizer rather than summarizing it here. In my experience I've found keeping the raw data useful as additional statistics can be computed before visualization or during visualization as we may not compute all the statistics on day 1.